apiserver/facades/agent/uniter/networkinfoiaas.go

```diff
@@ -9,18 +9,49 @@ import (
 
 	"github.com/juju/collections/set"
 	"github.com/juju/errors"
+	"github.com/juju/names/v4"
 
 	apiservererrors "github.com/juju/juju/apiserver/errors"
 	"github.com/juju/juju/apiserver/params"
 	"github.com/juju/juju/core/network"
 	"github.com/juju/juju/state"
 )
 
+// Machine describes methods required for interrogating
+// the addresses of a machine in state.
+type Machine interface {
+	// Id returns the machine's model-unique identifier.
+	Id() string
+
+	// MachineTag returns the machine's tag, specifically typed.
+	MachineTag() names.MachineTag
+
+	// PrivateAddress returns the machine's preferred private address.
+	PrivateAddress() (network.SpaceAddress, error)
+
+	// AllAddresses returns the state representation of a machine's addresses.
+	// TODO (manadart 2021-06-15): Indirect this too.
+	AllAddresses() ([]*state.Address, error)
+
+	// AllDeviceSpaceAddresses returns all known addresses
+	// from the machine's link-layer devices.
+	AllDeviceSpaceAddresses() (network.SpaceAddresses, error)
+}
+
+// machine shims the state representation of a machine in order to implement
+// the Machine indirection above.
+type machine struct {
+	*state.Machine
+}
+
 // NetworkInfoIAAS is used to provide network info for IAAS units.
 type NetworkInfoIAAS struct {
 	*NetworkInfoBase
 
-	// machineNetworkInfos container network info for the unit's machine,
+	// machine is where the unit resides.
+	machine Machine
+
+	// machineNetworkInfos contains network info for the unit's machine,
```

Comment:
There is a subtle difference between this explanation and the method comment header block for populateMachineNetworkInfos.  The method indicates that only the unit's bound spaces will be included, not clear here.

Reply:
Good catch.